RA_Integration/IRA_Dialog.h

```diff
@@ -0,0 +1,100 @@
+#ifndef RA_DIALOG_H
+#define RA_DIALOG_H
+#pragma once
+
+#include <wtypes.h>
+
+
+/// <summary>
+/// Base class for all RA Dialogs.
+/// Not a "true" interface as not all dialogs share this functionality, but needed
+/// for the Message Queue.
+/// </summary>
+/// <remarks>
+///		Functions that are always different are pure virtual, 
+///		functions that are always the same are not virtual, 
+///		functions that are ususally the same but can be different are just virtual.
+/// </remarks>
+class IRA_Dialog
+{
+public:
+	// I can't figure out another way to do this
+	// basically just for those "four" dialogs in RA_Core you have to put false
```

Comment:
Can't you just set `isModal` in `DoModal`? Why does the constructor need to know?

Reply:
It needs to know that because when WM_CLOSE is called, modals are destroyed using EndDialog and modeless are destroyed using DestroyWindow. Plus modals are created using "DialogBox" and modeless are created using "CreateDialog".

If  you don't do it this way, the OS will call the wrong function.

Reply:
yes, but does it need to be in the constructor? can't you just set it when you call `DoModal`?

Reply:
Alright, I guess that makes sense since it's already set to false in the default constructor.